crates/router/src/core/payment_methods.rs

```diff
@@ -140,11 +141,16 @@ pub async fn initiate_pm_collect_link(
     )?;
 
     // Return response
+    let url = pm_collect_link.url.peek();
     let response = payment_methods::PaymentMethodCollectLinkResponse {
         pm_collect_link_id: pm_collect_link.link_id,
         customer_id,
         expiry: pm_collect_link.expiry,
-        link: pm_collect_link.url,
+        link: url::Url::parse(url)
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable_lazy(|| {
+                format!("Failed to parse the payment method collect link - {}", url)
+            })?,
```

Comment:
If this URL is going to be an external input (say provided by the merchant), then raising an internal server error may not be suitable. If it is something that we generate, then it can be an internal server error.

Also, is making it a `Url` in the other places an option here?

Reply:
By other places, do you mean making it a `Url` in storage models?

I have kept the URL specific fields in storage models as a `String`. It's going to use `Url` in every place (API interface + data injection in HTML) other than storage models